src/terrain/terrain.cpp

```diff
@@ -330,11 +330,11 @@ Terrain::get_node_group(const Tile* t) {
 
 const NodeGroup*
 Terrain::get_node_group(int xyz) const {
-    try {
-        return tile_to_group_.at(xyz);
-    } catch (const std::out_of_range& e) {
+    auto out = tile_to_group_.find(xyz);
+    if (out == tile_to_group_.end()){
         return nullptr;
```

Comment:
A better option would be [`std::optional`](https://en.cppreference.com/w/cpp/utility/optional)

Reply:
Ill do this is a terrain realated pr

Reply:
Make an issue.